components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -210,51 +231,74 @@ def get_mysql_connection_params(self, disable_localhost_socket_connection: bool
-            "username": self.username,
-            "password": self.password,
-            "name": self.name,
-            "table_prefix": CLP_METADATA_TABLE_PREFIX,
+            "credentials": {user_type: self.credentials[user_type].model_dump()},
+            "database": self.name,
             "compress": self.compress,
             "autocommit": self.auto_commit,
+            "name": self.name,
+            "type": DatabaseEngine.MYSQL.value,
+            "table_prefix": CLP_METADATA_TABLE_PREFIX,
         }
-        if self.ssl_cert:
-            connection_params_and_type["ssl_cert"] = self.ssl_cert
-        return connection_params_and_type
 
     def dump_to_primitive_dict(self):
-        d = self.model_dump(exclude={"username", "password"})
+        d = self.model_dump(exclude={"credentials"})
         return d
 
     def load_credentials_from_file(self, credentials_file_path: pathlib.Path):
         config = read_yaml_config_file(credentials_file_path)
         if config is None:
             raise ValueError(f"Credentials file '{credentials_file_path}' is empty.")
         try:
-            self.username = get_config_value(config, f"{DB_COMPONENT_NAME}.username")
-            self.password = get_config_value(config, f"{DB_COMPONENT_NAME}.password")
+            self.credentials[ClpDbUserType.CLP].username = get_config_value(
+                config, f"{DB_COMPONENT_NAME}.username"
+            )
+            self.credentials[ClpDbUserType.CLP].password = get_config_value(
+                config, f"{DB_COMPONENT_NAME}.password"
+            )
+            self.credentials[ClpDbUserType.ROOT].username = get_config_value(
+                config, f"{DB_COMPONENT_NAME}.root_username"
+            )
+            self.credentials[ClpDbUserType.ROOT].password = get_config_value(
+                config, f"{DB_COMPONENT_NAME}.root_password"
+            )
         except KeyError as ex:
             raise ValueError(
                 f"Credentials file '{credentials_file_path}' does not contain key '{ex}'."
             )
 
-    def load_credentials_from_env(self):
+    def load_credentials_from_env(self, user_type: ClpDbUserType = ClpDbUserType.CLP):
         """
-        :raise ValueError: if any expected environment variable is not set.
+        Loads database credentials from environment variables.
+
+        :param user_type: User type whose credentials are to be loaded.
+
+        :raise ValueError: If any expected environment variable is not set.
         """
-        self.username = _get_env_var(CLP_DB_USER_ENV_VAR_NAME)
-        self.password = _get_env_var(CLP_DB_PASS_ENV_VAR_NAME)
+        match user_type:
+            case ClpDbUserType.CLP:
+                user_env_var = CLP_DB_USER_ENV_VAR_NAME
+                pass_env_var = CLP_DB_PASS_ENV_VAR_NAME
+            case ClpDbUserType.ROOT:
+                user_env_var = CLP_DB_ROOT_USER_ENV_VAR_NAME
+                pass_env_var = CLP_DB_ROOT_PASS_ENV_VAR_NAME
```

Comment:
IDEs are not smart enough to tell whether the strings are set even if we throw in the `match`
```
        user_env_var: str = ""
        pass_env_var: str = ""
        match user_type:
```

so i think it's better to write
```suggestion
        if user_type == ClpDbUserType.CLP:
            user_env_var = CLP_DB_USER_ENV_VAR_NAME
            pass_env_var = CLP_DB_PASS_ENV_VAR_NAME
        elif user_type == ClpDbUserType.ROOT:
            user_env_var = CLP_DB_ROOT_USER_ENV_VAR_NAME
            pass_env_var = CLP_DB_ROOT_PASS_ENV_VAR_NAME
        else:
            raise ValueError(f"Unsupported user type '{user_type}'.")
```